stonesoup/reader/base.py

```diff
@@ -74,3 +74,43 @@ def sensor_data_gen(self):
             Sensor data generated in the time step
         """
         raise NotImplementedError
+
+
+class FrameReader(SensorDataReader):
+    """FrameReader base class
+    A FrameReader produces :class:`~.SensorData` in the form of
+    :class:`~ImageFrame` objects.
+    """
+
+    @property
+    def frame(self):
+        return self.sensor_data
+
+    @abstractmethod
+    @BufferedGenerator.generator_method
+    def frames_gen(self):
+        """Returns a generator of frames for each time step.
+        Yields
+        ------
+        : :class:`datetime.datetime`
+            Datetime of current time step
+        : set of :class:`~.ImageFrame`
+            Generated frame in the time step
+        """
```

Comment:
Getting a warning on documentation build. `stonesoup/reader/base.py:docstring of stonesoup.reader.base.FrameReader.frames_gen:3: WARNING: Unexpected indentation.` May be lack of blank line before `Yields` heading